src/Dfe.PlanTech.Data.Sql/Repositories/SubmissionRepository.cs

```diff
@@ -239,6 +235,38 @@ public async Task<SubmissionEntity> SetSubmissionInaccessibleAsync(int submissio
         return submission;
     }
 
+    public async Task SetSubmissionInProgressAsync(
+        int establishmentId,
+        string sectionId)
+    {
+        var query = GetPreviousSubmissionsInDescendingOrder(establishmentId, sectionId, isCompletedSubmission: false);
+
+        var submission = await query.FirstOrDefaultAsync();
+        if (submission is null)
+        {
+            throw new InvalidOperationException($"Submission not found for establishment ID '{establishmentId}' and section ID '{sectionId}'");
+        }
+
+        await SetSubmissionInProgressAsync(submission.Id);
+    }
+
+    public async Task<SubmissionEntity> SetSubmissionInProgressAsync(int submissionId)
+    {
+        var submission = await GetSubmissionByIdAsync(submissionId);
+        if (submission is null)
+        {
+            throw new InvalidOperationException($"Submission not found for ID '{submissionId}'");
+        }
+
+        if (submission.Status == "Inaccessible")
```

Comment:
Similar to Jag's comment above, here's another place we might use the enum `SubmissionStatus.Inaccessible` - might be worth a scan through for any additional examples.